api/src/changeRequests/processChangeRequest.ts

```diff
@@ -17,18 +18,30 @@ export async function processChangeRequest(
         throw new Error(validationResult.error);
     }
 
-    const upsertResult = await db.upsertDoc(changeRequest.doc);
+    const upsertResult = await db.upsertDoc(validationResult.validatedData);
 
     // Generate change document and store in database
     if (upsertResult.changes) {
         const changeDoc = new ChangeDto();
         changeDoc._id = "change:" + randomUUID();
         changeDoc.type = DocType.Change;
-        changeDoc.docId = changeRequest.doc._id;
-        changeDoc.docType = changeRequest.doc.type;
+        changeDoc.docId = validationResult.validatedData._id;
+        changeDoc.docType = validationResult.validatedData.type;
         changeDoc.updatedTimeUtc = upsertResult.updatedTimeUtc;
         changeDoc.changes = upsertResult.changes;
+        changeDoc.changedByUser = userId;
+
+        // Apply the group membership or ACL of the included document to the change document
+        if (upsertResult.changes.memberOf) {
+            changeDoc.memberOf = upsertResult.changes.memberOf;
+        }
+
+        if (upsertResult.changes.acl) {
+            changeDoc.acl = upsertResult.changes.acl;
+        }
 
         return db.insertDoc(changeDoc);
+    } else {
+        return upsertResult;
     }
```

Comment:
Typical LG comment: the `else` is not needed since you already early return inside the `if` 😅